custom_components/tahoma/coordinator.py

```diff
@@ -50,16 +55,36 @@ async def _async_update_data(self) -> Dict[str, Device]:
             raise UpdateFailed(f"Error communicating with the TaHoma API: {exception}")
         else:
             for event in events:
+                if event.name == "DeviceAvailableEvent":
+                    self.devices[event.deviceurl].available = True
```

Comment:
When this event occurs?

Reply:
When the device status changes back to available, for example if it was without batteries or outside the network.